sdk/provisioning/Azure.Provisioning.CloudMachine/src/OFX/CloudMachineWorkspace.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using Azure.Core;
+using Azure.Identity;
+using Microsoft.Extensions.Configuration;
+
+namespace Azure.CloudMachine;
+
+public class CloudMachineWorkspace : WorkspaceClient
+{
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public string Id { get; }
+
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public override TokenCredential Credential { get; } = new ChainedTokenCredential(
+        new AzureCliCredential(),
+        new AzureDeveloperCliCredential()
+    );
+
+    [SuppressMessage("Usage", "AZC0007:DO provide a minimal constructor that takes only the parameters required to connect to the service.", Justification = "<Pending>")]
+    public CloudMachineWorkspace(DefaultAzureCredential? credential = default, IConfiguration? configuration = default)
```

Comment:
Nit - this should probably be `TokenCredential` instead of `DefaultAzureCredential`?